thorlcr/master/thgraphmanager.cpp

```diff
@@ -1545,15 +1550,25 @@ void thorMain(ILogMsgHandler *logHandler, const char *wuid, const char *graphNam
                                 break; // timeout/abort
                             // else - reject/ignore duff message.
                         }
-                        if (0 == currentGraphName.length()) // only ever true if !multiJobLinger
+
+                        // The following is true if no workunit/graph have been received
+                        // MORE: I think it should also be executed if lingerPeriod is 0
```

Comment:
I have created a separate jira to fix this (when Jake gets back)

Reply:
@ghalliday - lingerPeriod cannot be 1 (as defined by the values schema)

Reply:
ok.. agree, lingerPeriod is optional (min 1 if set), if it is not set (0).. it does look like it is not exiting correctly, checking.

Reply:
In fact, lingerPeriod cannot be 0. It defaults to 60 seconds, and the schema will not allow it to be set 0.
So the if (lingerPeriod) condition is misleading, it will always enter this block of code in practice.